soa-derive-internal/src/iter.rs

```diff
@@ -124,7 +125,7 @@ pub fn derive(input: &Input) -> TokenStream {
             /// Get an iterator over the
             #[doc = #ref_doc_url]
             /// in this vector
-            pub fn iter(&self) -> #iter_name {
+            pub fn iter(&self) -> <#name as ::soa_derive::SoAIter>::Iter {
```

Comment:
I don't understand why this is required here?

For readbility, I find referring to the type directly better, instead of going through associated types of a trait

Reply:
I was likely tying the type system in balloon animals at one point and that made the system happy, and after reverting that particular design, it remained happy with those aliases so they remained. I'll restored the generated names

Reply:
I think this still needs updating